src/lib/Formula.ts

```diff
@@ -144,103 +133,57 @@ class Conjunction extends Formula {
     }
 
-    public isTrue(v: Map<string, boolean | null>): boolean | null {
-        for (const form of this.conjuncts) {
-            if (!form.isTrue(v)) {
-                return false;
-            }
-        }
-        return true;
+    public isTrue(v: Valuation): boolean {
+        return this.conjuncts.every(form => form.isTrue(v));
     }
 
     public atoms(): Set<PredicateAtom> {
-        const atoms = new Set<PredicateAtom>();
-
-        for (const form of this.conjuncts) {
-            form.atoms().forEach(atom => atoms.add(atom));
-        }
-
-        return atoms;
+        return new Set(this.subfs().flatMap(form => Array.from(form.atoms())));
     }
-
+   
     public constants(): Set<string> {
-        const consts = new Set<string>();
-
-        for (const form of this.conjuncts) {
-            form.constants().forEach(constant => consts.add(constant));
-        }
-
-        return consts;
+        return new Set(this.conjuncts.flatMap(form => Array.from(form.constants())));
     }
 
     public predicates(): Set<string> {
-        const preds = new Set<string>();
-
-        for (const form of this.conjuncts) {
-            form.predicates().forEach(predicate => preds.add(predicate));
-        }
-
-        return preds;
+        return new Set(this.conjuncts.flatMap(form => Array.from(form.predicates())));
     }
 }
 
 class Disjunction extends Formula {
     private disjuncts: Formula[];
 
     constructor(disjuncts: Formula[]) {
-        super('|', ' \\lor ');
+        super('∨', '\\lor');
         this.disjuncts = disjuncts;
     }
 
     public subfs(): Formula[] {
         return this.disjuncts;
     }
 
-    public isTrue(v: Map<string, boolean | null>): boolean | null {
-        for (const form of this.disjuncts) {
-            if (form.isTrue(v)) {
-                return true;
-            }
-        }
-        return false;
+    public isTrue(v: Valuation): boolean {
+        return this.disjuncts.some(form => form.isTrue(v));
     }
 
     public atoms(): Set<PredicateAtom> {
-        const atoms = new Set<PredicateAtom>();
-
-        for (const form of this.disjuncts) {
-            form.atoms().forEach(atom => atoms.add(atom));
-        }
-
-        return atoms;
+        return new Set(this.subfs().flatMap(form => Array.from(form.atoms())));
     }
 
     public constants(): Set<string> {
-        const consts = new Set<string>();
-
-        for (const form of this.disjuncts) {
-            form.constants().forEach(constant => consts.add(constant));
-        }
-
-        return consts;
+        return new Set(this.disjuncts.flatMap(form => Array.from(form.constants())));
     }
 
     public predicates(): Set<string> {
-        const preds = new Set<string>();
-
-        for (const form of this.disjuncts) {
-            form.predicates().forEach(predicate => preds.add(predicate));
-        }
-
-        return preds;
+        return new Set(this.disjuncts.flatMap(form => Array.from(form.predicates())));
```

Comment:
Ako som pisal v predchadzajucej review, tieto metody mozete pomocou `this.subfs().flatMap(…)` implementovat uz v triede `Formula`. Ak mate spravne definovanu `subfs()`, tak jedina trieda, v ktorej musia byt `atoms()`, `predicates()` a `constants()` definovane specialne je `PredicateAtom`.